src/app/raster/raster.component.ts

```diff
@@ -18,32 +21,24 @@ interface RasterData {
         <aol-coordinate [x]="1.4886" [y]="43.5554" [srid]="'EPSG:4326'"></aol-coordinate>
       </aol-view>
 
-      <aol-layer-image *ngIf="selectLayer === 'osm'">
+      <aol-layer-image>
         <aol-source-raster
-          #raster1
-          [operation]="operation"
-          [threads]="threads"
-          [lib]="lib"
-          [operationType]="operationType"
-          (beforeOperations)="beforeOperations($event)"
-          (afterOperations)="afterOperations($event)"
-        >
-          <aol-source-osm></aol-source-osm>
-        </aol-source-raster>
-      </aol-layer-image>
-
-      <aol-layer-image *ngIf="selectLayer === 'xyz'">
-        <aol-source-raster
-          #raster2
           [operation]="operation"
           [threads]="threads"
           [lib]="lib"
           [operationType]="operationType"
           (beforeOperations)="beforeOperations($event)"
           (afterOperations)="afterOperations($event)"
         >
+          <aol-source-osm *ngIf="selectLayer === 'osm'" (tileLoadEnd)="tileLoadEnd$.next($event)"></aol-source-osm>
           <aol-source-xyz
-            url="https://c.tile.thunderforest.com/cycle/{z}/{x}/{y}.png?apikey=0e6fc415256d4fbb9b5166a718591d71"
+            *ngIf="selectLayer === 'xyz' || selectLayer === 'xyz2'"
+            (tileLoadEnd)="tileLoadEnd$.next($event)"
+            [url]="
```

Comment:
Can we simplify this? I don't think this solution scale well (more than 3 layers) and is not a good for an `example`.

Why the component doesn't hold an object like `{type:osm, url:'...'}` for each element? It should be a solution?

Reply:
I did this to test two cases. The first when we change completely source. The second and third when we only change properties in same source.

If I make an object it will re-instantiate XYZ source.

Reply:
I remove this example, and keep case in local branch.

Reply:
Ok after discussion.